### PR TITLE
Fix dynamic collection components

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**
Fix T870189

### CODE REVIEW

**Comment by IlyaKhD** [comment]
[T870189 Form - The entire form is updated on an attempt to update an item inside the group](https://trello.com/c/AK1nBeAc/2207-t870189-form-the-entire-form-is-updated-on-an-attempt-to-update-an-item-inside-the-group)

File: src/core/component.ts [hunk 1]
```diff
@@ -145,6 +123,19 @@ const BaseComponent: VueConstructor<IBaseComponent> = Vue.extend({
     },
 
     methods: {
+        $_applyConfigurationChanges(): void {
+            this.$_config.componentsCountChanged.forEach((component) => {
```
Comment (IlyaKhD) [inline src/core/component.ts:127]:
```suggestion
            this.$_config.componentsCountChanged.forEach(({optionPath, isCollection, removed }) => {
```

File: src/core/configuration-component.ts [hunk 2]
```diff
@@ -49,6 +49,20 @@ function initOptionChangedFunc(config, vueInstance: Pick<IVue, "$vnode" | "$prop
...
+function getComponentInfo(vueInstance: Pick<IVue, "$vnode">, removed?: boolean) {
+    const config = getConfig(vueInstance) as Configuration;
+    const name = config.name;
+    const parentPath =  config.ownerConfig && config.ownerConfig.fullPath;
+    const optionPath = parentPath ? `${parentPath}.${name}` : name;
+
+    return {
+        optionPath,
+        isCollection: config.isCollectionItem,
+        fullPath: config.fullPath,
```
Comment (IlyaKhD) [inline src/core/configuration-component.ts:61]: This field seems to be not used and can be removed

File: src/core/component.test.ts [hunk 3]
```diff
@@ -841,15 +841,85 @@ describe("nested option", () => {
     });
 
     it("remove nested component by condition", (done) => {
+        const nestedCollectionItem = buildTestConfigCtor();
```
[the remark this answers is not in the document]
Comment (dxvladislavvolkov) [reply src/core/component.test.ts:844]: I think that now the test is more correct than the old scenario

[the remark this answers is not in the document]
Comment (dxvladislavvolkov) [reply src/core/component.test.ts:844]: You can show or hidden components. But if want to change it to the same component with other options you should use v-if for the component too

[the remark this answers is not in the document]
Comment (dxvladislavvolkov) [reply src/core/component.test.ts:844]: We already have the test 
https://github.com/DevExpress/devextreme-vue/blob/master/src/core/component.test.ts#L867

[the remark this answers is not in the document]
Comment (IlyaKhD) [reply src/core/component.test.ts:844]: You're right, we have another test for that

File: src/core/configuration-component.ts [hunk 4]
```diff
@@ -49,6 +55,17 @@ function initOptionChangedFunc(config, vueInstance: Pick<IVue, "$vnode" | "$prop
     setEmitOptionChangedFunc(config, vueInstance, innerChanges);
 }
 
+function getComponentInfo({name, isCollectionItem, ownerConfig }: Configuration, removed?: boolean): IComponentInfo {
+    const parentPath =  ownerConfig && ownerConfig.fullPath;
+    const optionPath = parentPath ? `${parentPath}.${name}` : name || "";
```
Comment (IlyaKhD) [inline src/core/configuration-component.ts:60]: `name || ""` indicates that name can be `undefined` or smth like, right?
In this case ``${parentPath}.${name}`` is equivalent to `${parentPath}.` - is this trailing dot OK?

**Pull request merged**